x/pool-incentives/keeper/keeper.go

```diff
@@ -147,6 +181,25 @@ func (k Keeper) GetPoolGaugeId(ctx sdk.Context, poolId uint64, lockableDuration
 	return sdk.BigEndianToUint64(bz), nil
 }
 
+// GetNoLockGaugeIdsFromPool returns all the NoLock gauge ids associated with the pool id.
+// This can only be used for the "NoLock" gauges. For "NoLock" gauges there are 2 kinds
+// of links created:
+// - general
+// - by duration (for internal incentives)
+//
+// Every "NoLock" gauge has the first link. Only the internal incentives "NoLock" gauges
+// have the second link.
+func (k Keeper) GetNoLockGaugeIdsFromPool(ctx sdk.Context, poolId uint64) ([]uint64, error) {
```

Comment:
this function doesnot seem to be used anywhere. Are we writing a no-lock query in the future?

Reply:
Its used in keeper_test and gauge_test, but yeah I don't see it in the actual code only as test helper

Reply:
Yeah, unfortunately since this function is needed in incentives tests, I cannot make it test-specific or unexported.